### PR TITLE
Adds type checking to CI

### PR DESCRIPTION
**Pull request by scalvert**
Our current CI doesn't do any type checking, which incorrectly signals that everything is valid up until you try to release. During release, we run a build via `ember ts:precompile`, which will error if there are issues with types.

Adding this type check during CI will help avoid this situation.

Relates to #272, depends on #280

### CODE REVIEW

**Comment by scalvert** [comment]
CI will fail since we currently have a type mismatch due to #272

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -23,6 +23,8 @@ jobs:
         run: yarn install
       - name: lint
         run: yarn lint:js
+      - name: validate
+        run: yarn tsc --noEmit
```
Comment (rwjblue) [inline .github/workflows/ci-build.yml:27]: We already have a script for this `yarn lint:ts`

https://github.com/emberjs/ember-test-waiters/blob/1f4ed4d96c4e5958c4ab286f734ee052fd60e236/package.json#L20-L31

We should tweak to match the more modern addon blueprint for our scripts, so we run `yarn lint` in CI and that runs `lint:*` (along with `yarn test` running `yarn lint`, to ensure that local testers just running `yarn test` have a proper "pass"/"fail" experience).

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (scalvert) [reply .github/workflows/ci-build.yml:27]: Oh good call. I'll update to that.

Comment (scalvert) [reply .github/workflows/ci-build.yml]: OK, I've updated the scripts to more closely match the blueprint, though there are some small differences. Let me know if you think that makes sense.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**